tests/test_parquet.py

```diff
@@ -1019,7 +1033,22 @@ def _checkAstropyTableEquality(self, table1, table2, skip_units=False, has_bigen
                 self.assertEqual(table1[name].unit, table2[name].unit)
                 self.assertEqual(table1[name].description, table2[name].description)
                 self.assertEqual(table1[name].format, table2[name].format)
-        self.assertTrue(np.all(table1 == table2))
+                # We need to check masked/regular columns after filling.
+                has_masked = False
+                if isinstance(table1[name], atable.column.MaskedColumn):
+                    c1 = table1[name].filled()
+                    has_masked = True
+                else:
+                    c1 = np.array(table1[name])
+                if isinstance(table2[name], atable.column.MaskedColumn):
+                    c2 = table2[name].filled()
+                    has_masked = True
+                else:
+                    c2 = np.array(table2[name])
+                np.testing.assert_array_equal(c1, c2)
+                # If we have a masked column then we test the underlying data.
+                if has_masked:
+                    np.testing.assert_array_equal(np.array(c1), np.array(c2))
```

Comment:
You may end up casting to array twice if only one of c1/c2 is masked, but I suppose that's harmless.

Reply:
These are actually astropy columns, so there is a cast in both cases!  (But it's a fast no-copy cast which would happen anyway via the `__array__` interface).